core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -845,6 +852,13 @@ private[spark] object Utils extends Logging {
     customHostname.getOrElse(localIpAddressHostname)
   }
 
+  /**
+   * Get the local machine's URI.
+   */
+  def localHostURI(): String = {
```

Comment:
Got it. I'm wondering when you _wouldn't_ want to use the "URI safe" name? they're identical in almost every case. Hm. I suppose the safer thing is to only format for URIs where, well, creating URIs. I'll have to dig into this more but generally want to minimize the change here and narrowly address the formatting issue